### ISSUE TITLE
Paper Mario Depth Fighting

### ISSUE THREAD

**Issue by thecoreyburton** [gonetz/GLideN64#1904]
When playing Paper Mario (two years ago) I noticed several times throughout the game some things will be missing from the screen. It was most common when they're in roughly the same location. Sometimes they'd be entirely absent, other times they'd appear as several horizontal lines, or even a combination of both, flickering back and forth, like the two were fighting for depth. At the time, swapping RSP to cxd4 LLE seemed to fix the problem but caused many others.

The following screenshots are not my own, but were show to me and reminded me of this issue. The scene occurs at the end of Chapter 3, after you fight the heart and the creature is supposed to have a tongue. In the first shot, it's entirely absent. In the second shot, it's half showing through as horizontal lines.

![a](https://user-images.githubusercontent.com/12134368/45248950-27725400-b35b-11e8-83ed-8e347940edb5.png)
![b](https://user-images.githubusercontent.com/12134368/45248949-23decd00-b35b-11e8-836b-1bd9fb7f9994.png)

**Comment by theboy181** [comment]
Does the latest build have this issue still?

**Comment by thecoreyburton** [comment]
Yes. The the tongue is still absent, as are certain other similar elements in the game.

**Comment by olivieryuyu** [comment]
Can we have a savestate pls?

**Comment by gonetz** [comment]
Yes, save state needed. I can't run through the game to that point.

**Comment by thecoreyburton** [comment]
Here are two savestates:
[1.pj.zip](https://github.com/gonetz/GLideN64/files/2404002/1.pj.zip)
[2.pj.zip](https://github.com/gonetz/GLideN64/files/2404003/2.pj.zip)

I left it so that several presses of the A button should lead to the problem being demonstrated in either savestate.

In the first, the tongue should be visible when the heart jumps in to the mouth of Tubba Blubba (the monster, for anyone unfamiliar with the game). In the second, the tongue should be visible when the ghosts are released from his mouth.

**Comment by gonetz** [comment]
@thecoreyburton I see the problem now, thanks!

**Comment by gonetz** [comment]
It is regression. It happened when I migrated from VS 2010 to VS 2013: 4f8c3f691
There is no depth issues with previous commit d7f4e44674.
No changes in code, only PlatformToolset changed to v120 - depth compare does not work properly.
I don't know how to fix it.
GLideN64 for m64p built with GCC, but it has that depth compare issue too.

**Comment by LegendOfDragoon** [comment]
Maybe you could check warnings, by enabling /Wall
